### DISCUSSION TITLE
Error with java agent instrumenting java application

### DISCUSSION THREAD

**Discussion by SenthilkumarRasan** [apache/skywalking#8147] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

Tried skywalking java agent to use with our java springboot application. 
Following errors are blocking us to use skywalking

ERROR 2021-11-18 22:13:59:541 pool-3-thread-1 ConstructorInter : ConstructorInter failure.
java.lang.ClassCastException: class com.datastax.driver.core.TranslatedAddressEndPoint cannot be cast to class java.net.InetSocketAddress (com.datastax.driver.core.TranslatedAddressEndPoint is in unnamed module of loader org.springframework.boot.loader.LaunchedURLClassLoader @8c46918; java.net.InetSocketAddress is in module java.base of loader 'bootstrap')

Another error for tracing
WARN 2021-11-18 22:17:51:342 qtp423765194-51 TracingContext : More than 600 spans required to create
java.lang.RuntimeException: Shadow tracing context. Thread dump

### What you expected to happen

We are not getting stats and traces due to above errors

### How to reproduce

I can send snapshots or complete log files

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
> unnamed module of loader

What is `com.datastax.driver.core.TranslatedAddressEndPoint` for? Could you take a look at the stack and find out what is the hierarchy tree of this class? And which plugin reports this error? 

> Another error for tracing
WARN 2021-11-18 22:17:51:342 qtp423765194-51 TracingContext : More than 600 spans required to create
java.lang.RuntimeException: Shadow tracing context. Thread dump

This means some plugins don't run as expected, I guess the previous error may be a root cause.
____

BTW, which version are you using?

**Comment by SenthilkumarRasan** [comment]
I am using Skywalking 8.8.1.
datastax driver is used to connct to AWS cassandra DB.
Issue is i dont see any stats in dashboard section other than service/instance/endpoints
How do i get which driver is causing the error? Is there any log file in Skywalking OAP service to get more details
Started seeing few traces now. Still no stats for response time and others in dashboard page.

**Reply by wu-sheng** [reply]
Check plugins folder, there are many plugins in default. You could add one by one base on your requirements, and check which one causes this. 
Generally, if you are using Spring, keep Spring plugins only first.

**Reply by SenthilkumarRasan** [reply]
finally i was able to see data in sky-walking UI. It seems takes lot of time before data gets into OAP. Is there any logfiles in OAP to make sure all are working fine.

**Reply by wu-sheng** [reply]
SkyWalking Java agent is designed for server, not a simple call. I think you misused the agent.

**Reply by wu-sheng** [reply]
For server use case, there is no point to say finished.
And agent has its log file to output segments reported, if in debug level.

**Reply by SenthilkumarRasan** [reply]
I have skywalking OAP running in kubernetes with 2 replicas. Executed couple of load tests for java application with skywalking java agent. I dont see stats in skywalking UI even after an hour. It takes lots of time to show it in UI. Another question related to dashboards.. by mistake i have deleted endpoint dashboard and wasnt able to get that back. How do i add endpoint dashboard back to dashboards page

**Reply by wu-sheng** [reply]
Then, do you set utc0 on UI? It should automatically set. Also, is the time of your service pod or machine being set correctly? 
SkyWalking persistents metrics in 3-5 seconds top.

**Reply by SenthilkumarRasan** [reply]
Verified the time on k8 nodes. Looks good.  With default tracing of 100%, I don't see some of the traces and takes hours before traces appears in ui. Could you provide instruction on get endpoint dashboard back as I have deleted by mistake?

**Reply by wu-sheng** [reply]
It is better to check on the database side. Check the segment(table/index) in your database, the agent actually reports the data in seconds latency as mentioned. If all things work, you should find out time bucket and raw data in the segment table pretty soon.

**Comment by SenthilkumarRasan** [comment]
Will check that out. Could you provide steps to bring back deleted dashboard in dashboards page
